src/pieces/memberHandler.ts

```diff
@@ -35,6 +39,21 @@ async function memberUpdate(oldMember: GuildMember | PartialGuildMember, newMemb
 	}
 }
 
+async function memberRemove(member: GuildMember | PartialGuildMember): Promise<void> {
+	if (member.guild.id !== GUILDS.MAIN) return;
+
+	const dbMember: SageUser = await member.client.mongo.collection(DB.USERS).findOne({ discordId: member.id });
+
+	if (!dbMember) return;
+
+	dbMember.isVerified = false;
+	dbMember.discordId = '';
+	dbMember.roles = dbMember.roles.filter(role => role !== ROLES.VERIFIED && role !== ROLES.STAFF);
+	dbMember.isStaff = false;
+
+	await member.client.mongo.collection(DB.USERS).replaceOne({ discordId: member.id }, dbMember);
```

Comment:
See above comment about partial members. If the client isn't on a partial member, it can be passed in.

Reply:
All is well when I clear the cache and leave the guild, no errors. Definitely not cached, here's the member log 
<img width="440" alt="image" src="https://user-images.githubusercontent.com/54593801/148452862-340cf09c-9977-4c9d-b536-4c73f1d27c6d.png">

Reply:
If it works, it works.